### DISCUSSION TITLE
Prevent a pending REST API order from being moved to trash

### DISCUSSION THREAD

**Discussion by therealgilles** [woocommerce/woocommerce#47828] [General]
Is there a way to prevent a order pending payment created through REST API from being moved to trash if it remains unpaid? I know orders created from the admin interface do not get moved to trash, so I'm hoping there must be a way to do the same for REST API orders.

**Comment by opr** [comment]
Hi @therealgilles thanks for your question.

I wonder if using the [`woocommerce_cancel_unpaid_order`](https://wp-kama.com/plugin/woocommerce/hook/woocommerce_cancel_unpaid_order) filter would work for you? You can check each order before it gets cancelled and skip ones you want to keep.

Interestingly, the default value of this filter is `'checkout' === $order->get_created_via()` which should not evaluate to true in the case of Store API orders 🤔 

@senadir and @wavvves anything else you want to add, or any thoughts on the above?

**Reply by therealgilles** [reply]
@opr Really appreciate your response.

I was able to toggle this setting to avoid having unpaid orders getting moved to the trash automatically:

<img width="436" alt="Screenshot 2024-06-09 at 12 04 43 PM" src="https://github.com/woocommerce/woocommerce/assets/18076992/9c991b39-e9ce-4961-9ea5-8197029e3900">

WooCommerce's behavior is poorly documented when it comes to handling unpaid orders. I gather the filter above has to do with the stock holding period. It is unclear how the setting above interacts with it.

It would also be great to have tooltips for UI settings pointing to potential filters that can be used to modify the behavior programmatically.

**Reply by therealgilles** [reply]
BTW the "Personal Data Retention" section is unclear about whether it pertains to when a user account has been deleted or not, though I am guessing not.

The setting for Store Credit/Gift certificate is very unclear. I am guessing (again) that this has to do with either expired gift certificates or gift certificates that have already been used:

<img width="426" alt="Screenshot 2024-06-09 at 12 19 45 PM" src="https://github.com/woocommerce/woocommerce/assets/18076992/6059d7d7-ff9b-4c12-8239-c8200ffa5831">

and now I'm also guessing that this is added by an extension and not part of WooCommerce core.
